src/MongoDB.AspNetCore.OData/MongoEnableQueryAttribute.cs

```diff
@@ -58,6 +61,11 @@ public override IQueryable ApplyQuery(IQueryable queryable, ODataQueryOptions qu
 
         var ignoreQueryOptions = AllowedQueryOptions.Select | AllowedQueryOptions.Expand;
         queryable = queryOptions.ApplyTo(queryable, ignoreQueryOptions);
+
+        // Expression generated by the oData library for substring is not translatable by Mongo LINQ Provider
+        var newExpr = GetRewrittenExpression(queryable);
+        queryable = queryable.Provider.CreateQuery(newExpr);
```

Comment:
I would suggest to move this 2 lines into a new method called like `RewriteExpression`. It should return new queryable. So the code will looks like:
`queryable = RewriteExpression(queryable)`
This let us keep `ApplyQuery` method clear and easy to read, when implementation details goes into another methods.

Reply:
Done.